tests/integration/test_algorithms.py

```diff
@@ -78,8 +77,15 @@ def test_gaussian_gate_teleportation(setup_eng, pure):
         Fourier | q[3]
 
     state = eng.run(prog).state
-    cov1 = state.reduced_gaussian(2)[1]
-    cov2 = state.reduced_gaussian(3)[1]
+    
+    if state._basis == "gaussian":
```

Comment:
What about using eng.backend_name? If that changes, the pytest marks will have to be changed right? In which case, this could be changed at the same time.